.evergreen/run-tokio-tests.sh

```diff
@@ -15,4 +15,10 @@ FEATURE_FLAGS="zstd-compression,snappy-compression,zlib-compression"
 echo "cargo test options: --features $FEATURE_FLAGS ${OPTIONS}"
 
 RUST_BACKTRACE=1 cargo test --features $FEATURE_FLAGS $OPTIONS | tee results.json
-cat results.json | cargo2junit > results.xml
+cat results.json | cargo2junit > async-tests.xml
+RUST_BACKTRACE=1 cargo test sync --features sync $FEATURE_FLAGS $OPTIONS | tee sync-tests.json
+cat sync-tests.json | cargo2junit > sync-tests.xml
+RUST_BACKTRACE=1 cargo test --doc sync --features sync $FEATURE_FLAGS $OPTIONS | tee sync-doc-tests.json
```

Comment:
I'm not exactly sure what's wrong here, but this test invocation seems to have some formatting errors per the evg output: https://evergreen.mongodb.com/task_log_raw/mongo_rust_driver_tests__async_runtime~tokio_auth_and_tls~auth_and_tls_compressor~no_compression_os~ubuntu_18.04_arm64_test_5.0_replica_set_patch_3b3e6f91372dc174a7b654509aa9963042bcd9de_621d44d632f417468148eef2_22_02_28_21_55_35/0?type=T#L2142

What's weird is that it doesn't appear to happen every time :thinking: 
There's also a junit-report-merger error there too that seems unrelated but also worth looking into.